distributed/shuffle/tests/test_merge.py

```diff
@@ -81,30 +80,27 @@ async def test_basic_merge(c, s, a, b, how):
     B = pd.DataFrame({"y": [1, 3, 4, 4, 5, 6], "z": [6, 5, 4, 3, 2, 1]})
     b = dd.repartition(B, [0, 2, 5])
 
-    joined = hash_join(a, "y", b, "y", how)
+    joined = a.merge(b, left_on="y", right_on="y", how=how)
```

Comment:
We should ensure that we indeed use a hashjoin, otherwise this test is worthless.

Reply:
Looks like here it's a `Merge`, is that okay?
```python
(Pdb) p list(joined.optimize().walk())
[Merge(4266cea), df, df]
```

Reply:
No, only ``HashJoinP2P`` is equivalent to hash_join, but it might just be a repr thing. I would assert explicitly

Reply:
eh, not looking good, this fails:
```python
    from dask_expr._merge import HashJoinP2P
    assert any(isinstance(expr, HashJoinP2P) for expr in joined._expr.walk())
```
Looks like it's `Merge`.

```python
(Pdb) p list(joined._expr.walk())
[Merge(4fd85ee), df, df]
(Pdb) p type(list(joined._expr.walk())[0])
<class 'dask_expr._merge.Merge'>
```

Reply:
you have to call optimise before it can be a HashJoin

Reply:
:man_facepalming: ah, right. (https://github.com/dask/distributed/pull/8560/commits/a67ac1d6d4e8049364f3c7cac49870fbbd76120a)